pkg/kubecost/asset.go

```diff
@@ -1241,21 +1248,26 @@ func (d *Disk) add(that *Disk) {
 	d.Cost += that.Cost
 
 	d.ByteHours += that.ByteHours
+
+	if that.StorageClass != UnknownStorageClass {
+		d.StorageClass = that.StorageClass
+	}
```

Comment:
I'm not entirely sure if this is the right way to handle StorageClass in `add()`. If I'm adding something with StorageClass `a` to something with StorageClass `b`, the StorageClass of the result is probably neither, right?

Check out how we do `d.Properties().Merge(that.Properties())` above -- I think it could be a useful guide.

Reply:
I might be wrong but add is called by Insert which is called by functions NewAssetSet, AggregatedBy, accumulate and InsertRange(called in KCM).

Believe the first two and last one should be okay with this if statement, but i think the accumulate might be impacted:
If a disk storage class is changed at any point of time the latest one should be listed not the __unknown__ . For example we are accumulating over 7d and first 3d the storageClass for a disk was was `standard` and the next 4 days storageClass was `standard-rwo`. It should show `standard-rwo` and not unknown in my opinion. Correct me if i'm wrong?

Reply:
we can't change Storage class of a Persistent volume, I tired, you need to provision a new PV claim which results in new persistent volume name with another storage class.

Reply:
Let's say I create a PV named `PV1` with storage class `SC1`. I use it for a day, then decide I need a different storage class. I delete `PV1`, then create a new PV named `PV1` with storage class `SC2`. Is this allowed? If so...

I'm pretty sure AggregateBy/Accumulate/Insert will treat these as the same "thing" and so we'll end up in this storage class case. Because we're combining the data from two same-named PVs with different storage classes, the storage class of the resulting Disk is ambiguous.

This is essentially a rephrasing of my question here: https://github.com/opencost/opencost/pull/1352#discussion_r940393985.

@nikovacevic a second opinion here would be helpful, because I think we're close to blurring the lines between AssetProperties and Disk-specific properties here in a way that might interact oddly with aggby/etc.

Reply:
Yep, I think @michaelmdresser is correct -- we're going to want to keep the StorageClass only if they match. Otherwise, we'll want no StorageClass (i.e.`""`).

The simplest way to understand this is to ask:  what happens if you aggregate by "type"? You'll get results of `Node`, `Disk`, etc. And what will the storage class of that one, aggregated `Disk` row be? Well, if they all match, then you can persist that property through all the `add` calls. If more than one storage class is involved, then we'll need to wipe that field.

To be explicit, consider the two following scenarios:

Scenario 1
* Disk D1 has storage class SC1
* Disk D2 has storage class SC1
* Disk D3 has storage class SC1
* D1 + D2 + D3 = DT
* DT has storage class SC1

Scenario 2
* Disk D1 has storage class SC1
* Disk D2 has storage class SC1
* Disk D3 has storage class SC2
* D1 + D2 + D3 = DT
* DT has no storage class (i.e. "")

As written, I believe scenario 2 would result in DT.StorageClass = SC2, which would be incorrect.

Does that make sense? Let me know if I can help @avrodrigues5 !